test/tools/libtesteth/TestHelper.cpp

```diff
@@ -286,10 +285,30 @@ int64_t toPositiveInt64(const json_spirit::mValue& _v)
     default:
         cwarn << "Bad type for scalar: " << _v.type();
     }
+    return n;
+}
 
-    if (n < 0)
-        throw std::out_of_range{"unexpected negative value: " + std::to_string(n)};
-
+uint64_t toPositiveInt64(const json_spirit::mValue& _v)
+{
+    uint64_t n = 0;
+    switch (_v.type())
+    {
+    case json_spirit::str_type:
+    {
+        long long readval = std::stoll(_v.get_str(), nullptr, 0);
+        if (readval < 0)
+            throw std::out_of_range{
```

Comment:
I recommend not to throw standard exceptions, but the ones derived from them, we have `DEV_SIMPLE_EXCEPTION` macro to define them.